src/backend/aspen/api/authn.py

```diff
@@ -57,28 +75,91 @@ async def setup_userinfo(session: AsyncSession, auth0_user_id: str) -> Optional[
     return user
 
 
+class MagicLinkPayload(TypedDict):
+    user_id: str
+    expiry: str
+
+
+async def magic_link_payload(
```

Comment:
I moved the magic link validation/authentication to this module instead of doing everything directly in the magic link reading endpoint... I'm not sure if this is going to make things easier or not, but it allows us to keep all of the "different ways that we can authenticate a user" in one place, and choose which ones take precedence over others here 🤷‍♀️

Reply:
Makes sense